src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -178,43 +179,57 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
   }
 
   fetchCounts = async () => {
-    const { allowedTypes } = this.props;
+    const { allowedTypes, namespaceRegistry } = this.props;
     const { queryText, visibleTypes, visibleNamespaces } = parseQuery(this.state.activeQuery);
 
-    const filteredTypes = allowedTypes.filter(
-      (type) => !visibleTypes || visibleTypes.includes(type)
-    );
+    const filteredTypes = filterQuery(allowedTypes, visibleTypes);
+
+    const availableNamespaces = namespaceRegistry.getAll()?.map((ns) => ns.id) || [];
+
+    const filteredCountOptions = {
+      typesToInclude: filteredTypes,
+      searchString: queryText,
+    };
+
+    if (availableNamespaces?.length) {
```

Comment:
Based on line 187, `availableNamespaces` is always an array:
```suggestion
    if (availableNamespaces.length) {
```